### PR TITLE
Kafka codebase refactor

### PR DESCRIPTION
**Pull request by worryg0d**
The PR provides refactor of the Kafka codebase. Also there are some code cleanups, logs verboses, etc.

### CODE REVIEW

File: config/samples/clusters_v1beta1_cassandra.yaml [hunk 1]
```diff
@@ -3,21 +3,21 @@ kind: Cassandra
...
+  name: "bohdan-cassandra" #(immutable)
   version: "4.0.10" #(immutable)
   privateNetwork: false #(immutable)
   dataCentres:
     - name: "AWS_cassandra" #(mutable)
       region: "US_EAST_1" #(immutable)
-#      debezium:
-#        - kafkaVpcType: "VPC_PEERED" #(immutable)
-#          kafkaTopicPrefix: "test" #(immutable)
+      debezium:
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply config/samples/clusters_v1beta1_cassandra.yaml]: Done

File: controllers/clusters/kafka_controller.go [hunk 2]
```diff
@@ -526,7 +530,7 @@ func (r *KafkaReconciler) startClusterStatusJob(kafka *v1beta1.Kafka) error {
 }
 
 func (r *KafkaReconciler) newWatchStatusJob(k *v1beta1.Kafka) scheduler.Job {
-	l := log.Log.WithValues("component", "kafkaStatusClusterJob")
+	l := log.Log.WithValues("component", "kafkaStatusClusterJob", "clusterID", k.Status.ID)
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply controllers/clusters/kafka_controller.go]: Done

File: controllers/clusters/kafka_controller.go [hunk 3]
```diff
@@ -602,9 +599,23 @@ func (r *KafkaReconciler) newWatchStatusJob(k *v1beta1.Kafka) scheduler.Job {
 			}
 		}
 
+		equals := k.Spec.IsEqual(iKafka.Spec)
+
+		if equals && k.Annotations[models.ExternalChangesAnnotation] == models.True {
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply controllers/clusters/kafka_controller.go:607]: Done

File: controllers/clusters/helpers.go [hunk 4]
```diff
@@ -336,3 +336,18 @@ func handleExternalChanges[T any](
 
 	return models.ExitReconcile, nil
 }
+
+func autoHandleExternalChanges(c client.Client, r record.EventRecorder, obj Object) error {
```
Comment (ribaraka) [inline controllers/clusters/helpers.go:340]: reconcileExternalChanges

**Review by ribaraka — CHANGES_REQUESTED** [review]

**Review by testisnullus — APPROVED** [review]

File: controllers/clusters/helpers.go (hunk 4 above)
Comment (worryg0d) [reply controllers/clusters/helpers.go]: fixed

**Review by ribaraka — APPROVED** [review]

**Pull request merged**